### ISSUE TITLE
`Callback` throws a `TypeError` when checking a `callable` has variadic parameters

### ISSUE THREAD

**Issue by ziadoz** [sebastianbergmann/phpunit#5946]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 11.3.4
| PHP version         | 8.3.7
| Installation Method | Composer

#### Summary

I think [recent changes](https://github.com/sebastianbergmann/phpunit/commit/18911e55e9a18d473668406a246346bed4a8afdc) to the `Callback` class to handle variadic parameters are causing this type error:

```
TypeError: ReflectionFunction::__construct(): Argument #1 ($function) must be of type Closure|string, class@anonymous given
```

#### Current behavior

The `Callback` class accepts a `callable` but it passes that to `ReflectionFunction`, which only accepts a `Closure`.

#### How to reproduce

You can reproduce it by directly calling the `Callback` class with a callable and then calling the `isVariadic` method:

```
$callback = new Callback(new class {
    public function __invoke(): void
    {
    }
});
$callback->isVariadic();
```

#### Expected behavior

The callable should be converted to a closure before being passed to `ReflectionFunction`.

I do have a branch open that fixes the issue. If it looks acceptable I can open a PR for it: https://github.com/sebastianbergmann/phpunit/compare/main...ziadoz:phpunit:fix-callback-callable-variadic?expand=1

**Comment by kubawerlos** [comment]
The changes on your branch look very good.

**Comment by ziadoz** [comment]
@kubawerlos Thanks. I've opened a PR with the fix: https://github.com/sebastianbergmann/phpunit/issues/5946

**Comment by sebastianbergmann** [comment]
Thank you.

In the future, please target the appropriate branch for bug fixes. I have cherry-picked your commits to `10.5` and merged them to `11.3` and `main` from there.
